trieste/acquisition/function/function.py

```diff
@@ -98,9 +98,17 @@ class ExpectedImprovement(SingleModelAcquisitionBuilder[ProbabilisticModel]):
     of the posterior mean at observed points.
```

Comment:
need to update the docstring to explain the `predictive_mean_fn` trick?

Reply:
I have added some text. Hopefully it is clear.

Reply:
all good